pkg/trait/service_binding.go

```diff
@@ -0,0 +1,266 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trait
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "github.com/apache/camel-k/pkg/apis/camel/v1"
+	sb "github.com/redhat-developer/service-binding-operator/pkg/apis/operators/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// The Service Binding trait allows users to connect to Provisioned Services and ServiceBindings in Kubernetes:
+// https://github.com/k8s-service-bindings/spec#service-binding
+// As the specification is still evolving this is subject to change
+// +camel-k:trait=service-binding
+type serviceBindingTrait struct {
+	BaseTrait `property:",squash"`
+	// List of Provisioned Services and ServiceBindings in the form KIND.VERSION.GROUP/NAME[/NAMESPACE]
+	ServiceBindings []string `property:"service-bindings" json:"serviceBindings,omitempty"`
+}
+
+func newServiceBindingTrait() Trait {
+	return &serviceBindingTrait{
+		BaseTrait: NewBaseTrait("service-binding", 250),
+	}
+}
+
+// IsAllowedInProfile overrides default
+func (t *serviceBindingTrait) IsAllowedInProfile(profile v1.TraitProfile) bool {
+	return profile == v1.TraitProfileKubernetes ||
+		profile == v1.TraitProfileOpenShift
```

Comment:
Is there a reason why the 3rd profile is excluded (Knative)?

Reply:
No, removed